Framework/MDAlgorithms/src/MDNorm.cpp

```diff
@@ -1136,13 +1139,165 @@ void MDNorm::cacheDimensionXValues() {
+  auto intersectionsBegin = intersections.begin();
+  // copy momenta to xValues
+  xValues.resize(intersections.size());
+  yValues.resize(intersections.size());
+  auto x = xValues.begin();
+  for (auto it = intersectionsBegin; it != intersections.end(); ++it, ++x) {
+    *x = (*it)[3];
+  }
+  // calculate integrals at momenta from xValues by interpolating between
+  // points in spectrum sp
+  // of workspace integrFlux. The result is stored in yValues
+  calcIntegralsForIntersections(xValues, integrFlux, wsIdx, yValues);
+  // [VZ QUESTION] Shall background be duplicated here? (Answer: No)
+}
+
+/**
+ * Calculate the normalization among intersections on a single detector
+ * in 1 specific SpectrumInfo/ExperimentInfo
+ * @param intersections: intersections
+ * @param solid: proton charge
+ * @param yValues: original signal values
+ * @param vmdDims: MD dimensions
+ * @param pos: positions from intersections for memory efficiency
+ * @param posNew: transformed positions
+ * @param signalArray: (output) signals
+ */
+inline void MDNorm::calcSingleDetectorNorm(
+    const std::vector<std::array<double, 4>> &intersections,
+    const double &solid, std::vector<double> &yValues, const size_t &vmdDims,
+    std::vector<coord_t> &pos, std::vector<coord_t> &posNew,
+    std::vector<std::atomic<signal_t>> &signalArray) {
+
+  auto intersectionsBegin = intersections.begin();
+  for (auto it = intersectionsBegin + 1; it != intersections.end(); ++it) {
+
+    const auto &curIntSec = *it;
+    const auto &prevIntSec = *(it - 1);
+
+    // The full vector isn't used so compute only what is necessary
+    // If the difference between 2 adjacent intersection is trivial, no
+    // intersection normalization is to be calculated
+    double delta, eps;
+    if (m_diffraction) {
+      // diffraction
+      delta = curIntSec[3] - prevIntSec[3];
+      eps = 1e-7;
+    } else {
+      // inelastic
+      delta = (curIntSec[3] * curIntSec[3] - prevIntSec[3] * prevIntSec[3]) /
+              energyToK;
+      eps = 1e-10;
+    }
+    if (delta < eps)
+      continue; // Assume zero contribution if difference is small
+
+    // Average between two intersections for final position
+    // [Task 89] Sample and background have same 'pos[]'
+    std::transform(curIntSec.data(), curIntSec.data() + vmdDims,
+                   prevIntSec.data(), pos.begin(),
+                   [](const double rhs, const double lhs) {
+                     return static_cast<coord_t>(0.5 * (rhs + lhs));
+                   });
+    signal_t signal;
+    // [Task 89 signal_t bkgdSignal;
+    if (m_diffraction) {
+      // Diffraction
+      // index of the current intersection
+      auto k = static_cast<size_t>(std::distance(intersectionsBegin, it));
+      // signal = integral between two consecutive intersections
+      signal = (yValues[k] - yValues[k - 1]) * solid;
+      // [Task 89] signalBkgd = (bkgdValues[k] - bkgdValues[k - 1]) * solidBkgd;
```

Comment:
There won't be bkgdValues... yValues are from intersection integral with flux.  It is only to with Q values (x values of Worksapce2D flux workspace)

Reply:
signalBkgd = (yValues[k] - yValues[k - 1]) * solidBkgd;